sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -643,6 +644,14 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
     withTable("t") {
       val provider = "org.apache.spark.sql.test.DefaultSource"
       sql(s"CREATE TABLE t USING $provider")
+
+      // make sure the data source doesn't provide `InsertableRelation`, so that we can only append
+      // data to it with `CreatableRelationProvider.createRelation`
```

Comment:
One last comment. Let's explicitly say that we want to test the case that a data source is a CreatableRelationProvider but its relation does not implement InsertableRelation.